### ISSUE TITLE
problems with pip install on Debian Jessie

### ISSUE THREAD

**Issue by ultradian** [ipython-contrib/jupyter_contrib_nbextensions#785]
I am a new notebook user and installed using pip:

`pip install https://github.com/ipython-contrib/jupyter_contrib_nbextensions/tarball/master`

Everything looked good and the Nbextensions tab showed up, but it only had two entries: `Nbextensions dashboard tab` and `Nbextensions edit menu item`.  Investigating further, I found that all the nbextensions had been saved by default to `/usr/local/lib/python2.7/dist-packages/jupyter_contrib_nbextensions/nbextensions`.  I fixed it by creating a symlink in `~/.local/share/jupyter`, but I thought you might find this helpful for other people who are using Debian stable.  Perhaps the `jupyter contrib nbextension install --user` might be adjusted to accommodate this?

Thanks for all the work on a great package.

**Comment by jcb91** [comment]
Hi, thanks for flagging this. So, from what you describe, it seems like the contrib repo didn't install correctly at all - the path you mention, `/usr/local/lib/python2.7/dist-packages/jupyter_contrib_nbextensions/nbextensions`, is simply the python pip package's copy of the nbextensions files, from which the command `jupyter contrib nbextension install --user` should copy them to the user's jupyter data directory. From [jupyter_core/paths.py#L65-L69](https://github.com/jupyter/jupyter_core/blob/7e944323082f66f345736b68dad44f8a3691a812/jupyter_core/paths.py#L65-L69) we can see that this should be `~/.local/share/jupyter` on unix, provided the `XDG_DATA_HOME` environment variable is unset. So, if they weren't there, the command failed in some way. However, the fact that the `jupyter_nbextensions_configurator` got enabled suggests that you ran the command (unless you installed & enabled the configurator separately perhaps?). Could you post the output of the install command with the debug flag please? (you should at least temporarily remove/rename your symlink before trying this, to avoid problems) To be clear, the command I'd like the output of is

    jupyter contrib nbextension install --debug --user

which should give a lot of output about where files are getting copied (maybe pipe it into a file?). It's also possible that #774 has introduced a bug somewhere in this regard...

**Comment by ultradian** [comment]
Hmm.  You are correct that I ran the instruction  `jupyter contrib nbextension install --user` to install and get the configurartor working. Per your request, I renamed my symlink and ran `jupyter contrib nbextension install --debug --user`.  It worked flawlessly and created and copied all the files to `~/.local/share/jupyter/nbextensions` (so I didn't include the output).  I uninstalled and reinstalled a few more times and it worked each time.  I can't seem to recreate the problem.  Do you have any suggestions?

**Comment by jcb91** [comment]
It seems a little unlikely, but really the only thing I can think of is that perhaps you ran the analogous command for the _configurator_ only, which would be

    jupyter nbextensions_configurator enable --user

rather than the command for the whole contrib repo, which would be

    jupyter contrib nbextensions install --user

(or potentially with the `--system` or `--sys-prefix` flags would also get the same result). The only other option I can think of that achieves the result that you saw would be if you included the `--only-config` flag, which edits config files for install, but doesn't copy any files over.

The two nbextensions you mentioned (`Nbextensions dashboard tab` and `Nbextensions edit menu item`) are both provided directly by the `jupyter_nbextensions_configurator`, which injects its own entry into the notebook server's `nbextensions_path` attribute, so they're always available when using the configurator.

**Comment by ultradian** [comment]
Yes, I remember seeing the `notebook.json `file for those two entries when I first hit this issue.  I seem to recall finding those two files and no other extensions when looking in the usual places, bu now I can't seem to recreate or find any of that.  Sorry.  I know I did use the command `jupyter contrib nbextension install --user` but I can only assume that the last time I logged off it corrected something since nothing else was different from the first time I ran it.  Thank for all your effort in looking at this.  I'll close it up.
